runtime/runtime_impl_jetty9/src/main/java/com/google/apphosting/runtime/jetty9/JettyHttpHandler.java

```diff
@@ -0,0 +1,290 @@
+import com.google.apphosting.runtime.BackgroundRequestCoordinator;
+import com.google.apphosting.runtime.AppEngineConstants;
+import com.google.apphosting.runtime.LocalRpcContext;
+import com.google.apphosting.runtime.RequestManager;
+import com.google.apphosting.runtime.RequestRunner;
+import com.google.apphosting.runtime.RequestRunner.EagerRunner;
+import com.google.apphosting.runtime.ResponseAPIData;
+import com.google.apphosting.runtime.ServletEngineAdapter;
+import com.google.common.flogger.GoogleLogger;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.time.Duration;
+import java.util.concurrent.TimeoutException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.eclipse.jetty.server.Handler;
+import org.eclipse.jetty.server.Request;
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.server.handler.HandlerWrapper;
+
+/**
+ * This class replicates the behaviour of the {@link RequestRunner} for Requests which do not come
+ * through RPC. It should be added as a {@link Handler} to the Jetty {@link Server} wrapping the
+ * {@code AppEngineWebAppContext}.
+ *
+ * <p>This uses the {@link RequestManager} to start any AppEngine state associated with this request
+ * including the {@link ApiProxy.Environment} which it sets as a request attribute at {@link
+ * AppEngineConstants#ENVIRONMENT_ATTR}. This request attribute is pulled out by {@code
+ * ContextScopeListener}s installed by the {@code AppVersionHandlerFactory} implementations so that
+ * the {@link ApiProxy.Environment} is available all threads which are used to handle the request.
+ */
+public class JettyHttpHandler extends HandlerWrapper {
+  private static final GoogleLogger logger = GoogleLogger.forEnclosingClass();
+
+  private final boolean passThroughPrivateHeaders;
+  private final AppInfoFactory appInfoFactory;
+  private final AppVersionKey appVersionKey;
+  private final AppVersion appVersion;
+  private final RequestManager requestManager;
+  private final BackgroundRequestCoordinator coordinator;
+
+  public JettyHttpHandler(
+      ServletEngineAdapter.Config runtimeOptions,
+      AppVersion appVersion,
+      AppVersionKey appVersionKey,
+      AppInfoFactory appInfoFactory) {
+    this.passThroughPrivateHeaders = runtimeOptions.passThroughPrivateHeaders();
+    this.appInfoFactory = appInfoFactory;
+    this.appVersionKey = appVersionKey;
+    this.appVersion = appVersion;
+
+    ApiProxyImpl apiProxyImpl = (ApiProxyImpl) ApiProxy.getDelegate();
+    coordinator = apiProxyImpl.getBackgroundRequestCoordinator();
+    requestManager = (RequestManager) apiProxyImpl.getRequestThreadManager();
+  }
+
+  @Override
+  public void handle(
+      String target, Request baseRequest, HttpServletRequest request, HttpServletResponse response)
+      throws IOException, ServletException {
+
```

Comment:
Should there be support here for running in a mixed mode?  i.e. if the request is an instance of the RPC request class, should this handle become a noop?    Maybe even just an assert to make it clear that mode is not supported.

Reply:
I would prefer a flag that picks one or the other. We gonna remove rpc mode when we delete java8 support.

Reply:
Well in RPC mode we will never install this handler in the first place. 
Don't think its necessary, but I could something like 
```java
if (!Boolean.getBoolean(HTTP_CONNECTOR_MODE))
    throw new IllegalStateException();
```